spark-bigquery-dsv2/spark-bigquery-dsv2-common/src/main/java/com/google/cloud/spark/bigquery/v2/context/BigQueryDataSourceReaderContext.java

```diff
@@ -531,4 +560,11 @@ private long getRowSize(Collection<StructField> fields) {
             .map(structField -> (long) structField.dataType().defaultSize())
             .reduce(0L, Long::sum);
   }
+
+  public String getReadSessionId() {
+    if (readSessionResponse.isCalculated()) {
```

Comment:
Why not use `isBuilt` to check this?

Reply:
Didn't notice that. Fixed.

Reply:
Actually, we can't rely on `isBuilt` because it's set to true just when the create is submitted to the executor service so it might not be built.

Reply:
That's ok. We can call readSession.get after `isBuilt` is true. It's thread safe.

Reply:
It doesn't work as expected in integration tests and try to compute the read session twice. Example test : `testPartition_Daily`